zio-http/shared/src/main/scala/zio/http/codec/PathCodec.scala

```diff
@@ -310,6 +352,26 @@ sealed trait PathCodec[A] { self =>
     _optimize
   }
 
+  private def fallbacks(f: Fallback[_]): Set[String] = {
+    var fallbacks = Chunk(f.left, f.right)
+    val result    = ChunkBuilder.make[String]()
+    while (fallbacks.nonEmpty) {
```

Comment:
`head` / `tail` with `Chunk` is `O(n)`. And assuming `fallbacks` is called in compilation phase (which looks like it is), then no need for `while` loops as you can use more idiomatic, safer, nicer methods of doing the compilation.

Reply:
Done